### PR TITLE
fix modify user, offline save and load controller

### PR DESCRIPTION
**Pull request by tnguyen10**
Modifyuser: userid can't be changed, userid is passed instead of position
Offline save and load controller (affects Sign Up): changes user list to patient and provider list 

Bug: Sign Up, offline and online database are not synced updated -> bugs in modifyUserControlelr.getUser, since getUsers list does not have the signed up user

Some added features: problems are now tie to unique user id, fixed all suggestions from Michael

### CODE REVIEW

**Comment by modelarious** [comment]
Are we doing offline signup?  I think this is the one part that should be online only as two people (one offline and one online) could choose the same user ID and start making problems/Patient records for themselves.  Then when we come back online, we have to deal with two users with the same UserID.

In short, offline signup violates the requirement that user IDs must be unique.

File: app/src/main/java/Controllers/SignUpController.java [hunk 1]
```diff
@@ -37,11 +37,11 @@ public static void addProvider(String UserID, String Email, String Phone, Contex
...
-            ArrayList<User> users = new ArrayList<>();
+            ArrayList<Provider> users = new ArrayList<>();
             users.addAll(providers);
             
             //add new provider
-            users.add((User) provider);
+            users.add((Provider) provider);
             
             //commit to offline storage
             new OfflineSaveController().saveProviderList(users, context);
```
Comment (modelarious) [inline app/src/main/java/Controllers/SignUpController.java]: Probably could change this code to 
```
//get all the providers
ArrayList<Provider> providers = new 
        ElasticsearchProviderController.GetProviderTask().execute().get();

//add new provider
providers.add(provider);
//commit to offline storage
new OfflineSaveController().saveProviderList(users, context);

new ElasticsearchProviderController.AddProviderTask().execute(provider);

File: app/src/main/java/Controllers/SignUpController.java [hunk 2]
```diff
@@ -81,11 +81,11 @@ public static void addPatient(String UserID, String Email, String Phone, Context
                     ElasticsearchPatientController.GetPatientTask().execute().get();
 
             //add them to an array of users
-            ArrayList<User> users = new ArrayList<>();
+            ArrayList<Patient> users = new ArrayList<>();
             users.addAll(patients);
 
             //add new patient
-            users.add((User) patient);
+            users.add((Patient) patient);
```
Comment (modelarious) [inline app/src/main/java/Controllers/SignUpController.java]: Probably could change this code to 
```
//get all the patients
ArrayList<Patient> patients = new 
        ElasticsearchPatientController.GetPatientTask().execute().get();

//add new patient
patients.add(patient);

//commit to offline storage
new OfflineSaveController().savePatientList(users, context);

//commit to online storage
new ElasticsearchPatientController.AddPatientTask().execute(patient);
```

File: app/src/main/java/Controllers/SignUpController.java (hunk 1 above)
Comment (tnguyen10) [reply app/src/main/java/Controllers/SignUpController.java]: fixed this in new pull request

File: app/src/main/java/Controllers/SignUpController.java (hunk 2 above)
Comment (tnguyen10) [reply app/src/main/java/Controllers/SignUpController.java]: fixed this in new pull request

**Comment by tnguyen10** [comment]
I've changed patient and provider home menu test for it to pass
About pulling all datas from online to offline, this could be further discussed in the offline saving issue. But I think we do need to store them all because we don't know which user they are going to access

**Comment by tnguyen10** [comment]
I'm not sure why the copy righted files are remove. I'l try to make a new pull request after pulling everything from master

**Comment by tnguyen10** [comment]
I added some more codes for browsing problems, if anyone could take a look that would be great

**Comment by modelarious** [comment]
Still need the changes to the copyright profiles reverted.

**Comment by modelarious** [comment]
You could just add them back manually if you wanted to

**Comment by modelarious** [comment]
Addressing the offline storage of users: sure a provider might access any patient, but a patient only needs reference to itself as it has no use for other patients. I think it's fine for now and I won't make you change it.

**Comment by tnguyen10** [comment]
I've re-added the copy right files

**Pull request merged**